pkg/cableengine/healthchecker/healthchecker.go

```diff
@@ -0,0 +1,155 @@
+package healthchecker
+
+import (
+	"sync"
+	"time"
+
+	"github.com/go-ping/ping"
+	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	v1typed "github.com/submariner-io/submariner/pkg/client/clientset/versioned/typed/submariner.io/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+)
+
+var HealthCheckInterval = 15 * time.Second
+
+type HealthChecker struct {
+	localEndpointName string
+	client            v1typed.GatewayInterface
+	latencyMap        *sync.Map
+}
+
+type LatencyInfo struct {
+	Status  v1.ConnectionStatus
+	Latency *v1.LatencySpec
+}
+
+// NewEngine creates a new Engine for the local cluster
+func NewHealthChecker(localEndpointName string, client v1typed.GatewayInterface,
+	latencyMap *sync.Map) *HealthChecker {
+	return &HealthChecker{
+		localEndpointName: localEndpointName,
+		client:            client,
+		latencyMap:        latencyMap,
+	}
+}
+
+func (h *HealthChecker) Run(stopCh <-chan struct{}) {
+	go func() {
+		wait.Until(h.checkRemoteGatewayHealth, HealthCheckInterval, stopCh)
+	}()
+
+	klog.Infof("CableEngine healthchecker started")
+}
+func (h *HealthChecker) checkRemoteGatewayHealth() {
+	existingGw, err := h.client.Get(h.localEndpointName, metav1.GetOptions{})
```

Comment:
Instead of reading/writing from the API, I think it should be better to integrate the sync module here:

https://github.com/submariner-io/submariner/blob/master/pkg/cableengine/syncer/syncer.go

With the heatlhchecker, so we only write it once with the existing pinger status, while keeping pingers running in the background all the time.

Reply:
Done.